fix: query timeout kill query directed to incorrect instance (#730)
awslabs/aws-advanced-jdbc-wrapper: docs/using-the-jdbc-driver/using-plugins/UsingTheReadOnlyEndpointConnectionPlugin.md

# Read Only Endpoint Connection Plugin

> When making connections to the reader cluster endpoint of an Aurora database, each connection is made to a random available reader instance. This may cause errors if operations depend on making connections to the same instance. For example, to execute a query timeout with the MySQL Connector/J 8.0 Driver, a new internal connection will be made to kill the initial query. If the reader endpoint is used, the kill query may fail to cancel the initial query because the queries will be made to different hosts.

Reviewer: Can we make clear distinction between reader instance endpoint and reader cluster endpoints?

Reviewer: Can we draw more connection on how the plugin resolves the problem?

> When making connections to the reader cluster endpoint of an Aurora database, each connection is made to a random available reader instance. This may cause errors if operations depend on making connections to the same instance. For example, to execute a query timeout with the MySQL Connector/J 8.0 Driver, a new internal connection will be made to kill the initial query. If the reader endpoint is used, the kill query may fail to cancel the initial query because the queries will be made to different hosts.

Reviewer: Can we make it clear that this plugin doesn't support TAZ?

> When the Read Only Endpoint Connection Plugin is enabled and a connection to the read only endpoint is made, the connection will be made to a random reader. Before that connection is return, the Read Only Endpoint Connection Plugin will query for the current instance name and will create a new connection to that instance. The first connection that was made using the read only endpoint will be replaced by the plugin with the instance endpoint and the instance connection will be returned to the user.

Reviewer: ?
```suggestion
When establishing connections to a read-only cluster endpoint using the Read Only Endpoint Connection Plugin, the plugin will connect to a random reader instance and return an instance endpoint.
```

## Prerequisites

> This plugin should only be used when connecting to an Aurora database with a read only endpoint.

Reviewer: ```suggestion
This plugin should only be used when connecting to an Aurora database with a read-only cluster endpoint.
```

## Enabling the Read Only Endpoint Connection Plugin

> To enable the Reade Only Endpoint Connection Plugin, add the plugin code `readOnlyEndpoint` to the [`wrapperPlugins`](../UsingTheJdbcDriver.md#connection-plugin-manager-parameters) value, or to the current [driver profile](../UsingTheJdbcDriver.md#connection-plugin-manager-parameters).

Reviewer 2: ```suggestion
To enable the Read Only Endpoint Connection Plugin, add the plugin code `readOnlyEndpoint` to the [`wrapperPlugins`](../UsingTheJdbcDriver.md#connection-plugin-manager-parameters) value, or to the current [driver profile](../UsingTheJdbcDriver.md#connection-plugin-manager-parameters).
```
